### ISSUE TITLE
IApplicationBuilder does not contain a definition for UseSignalR

### ISSUE THREAD

**Issue by boombang** [aspnet/SignalR#491]
Today update Microsoft.NETCore.App and some v1.0.0-preview2-25241 version packages like Microsoft.AspNetCore.SignalR. And in `Startup.cs` : 
before
`public void ConfigureServices(IServiceCollection services)
{
  services.AddSignalR();
}
`
after
`public void ConfigureServices(IServiceCollection services)
{
  services.AddSignalRCore();
}
`
and in `public void Configure` function at `app.UseSignalr()` error 

> IApplicationBuilder” does not contain a definition for “UseSignalR

How to setup hub now?

**Comment by moozzyk** [comment]
You need to reference the [Microsoft.AspNetCore.SignalR.Http](https://dotnet.myget.org/feed/aspnetcore-ci-dev/package/nuget/Microsoft.AspNetCore.SignalR.Http) package now.

**Comment by boombang** [comment]
It works! Thank you!

**Comment by grahamehorner** [comment]
Q: I understand why this moved, as the IApplicationBuilder UseSignalR is/was http based, however shouldn't UseSignalR be transport agnostic? in that regardless of http/tcp/UDP etc. the UseSignalR is the same? and determines the availability of transport via dependency injection and/or configuration action?

**Comment by davidfowl** [comment]
No, IApplicationBuilder is http specific.

**Comment by grahamehorner** [comment]
@davidfowl Q: so if as a developer I was creating a web application that used SignalR with TCP as its transport the UseSignalR wouldn't need to be called ?

**Comment by davidfowl** [comment]
1. We don't have a TCP host yet (kestrel will be it though in case you were wondering post 2.0)
2. You wouldn't be configuring TCP things in the http pipeline (IApplicationBuilder)

**Comment by grahamehorner** [comment]
@davidfowl ok, so hyperthetically if I wished to use SignalR on multiple transports I'd need to call a method on each transport pipeline, and these extension method for this would be different for each transport type? do understand this is well in the furture and SignalR hasn't be released as yet plus the team has some Big Rocks to move.

**Comment by davidfowl** [comment]
This is correct. You need to configure it in each pipeline. The first class UseSignalR is a shortcut for http scenarios because it's the most common.  The MapSocket extension method exposes an ISocketBuilder per route which is all you really need

**Comment by matthiasburger** [comment]
Took me 8 hours to find out, that I need to add `Microsoft.AspNetCore.SignalR.Http` to my project. Is this documented anywhere?

Plus, I only could find it as Prerelease `1.0.0-alpha1-26873` ... I'm using .net core 2.1. Am I still on the right track? Did I miss something? Version 1.0.2 is on nuget.org, but not listed in my nuget package manager. Strange thing....

**Comment by BrennanConroy** [comment]
This is a very old issue with irrelevant info on it now.

I would suggest looking at our docs to get started: https://docs.microsoft.com/en-us/aspnet/core/tutorials/signalr?view=aspnetcore-2.1&tabs=visual-studio

If you are still having issues please open a new issue as we don't monitor closed issues as much as open ones.

**Comment by matthiasburger** [comment]
@BrennanConroy thanks for your reply - I found out what mistake I did. I used `Microsoft.AspNet.SignalR.*` packages instead of `Microsoft.AspNetCore.SignalR.*` packages. However, which NuGet-packages to use is not metioned in your documentations? When searching for SignalR I always get the old packages on top (I'm using JetBrains Rider IDE). 
Maybe, mentioning the package-names in the documentation would be great. Not only the client-things with npm but also the NuGets.

Didn't want to open a new issue for just telling my result now :)
